### PR TITLE
Added material design to closeAllTabs FAB icon

### LINKED ISSUE #1040: Material Design for CloseAllTab FAB not followed
**Describe the bug**
Material design is not followed in the CloseAllTabs FAB. Material design says 
*An app’s color scheme determines its color fill, which should contrast with the area behind the FAB.* 
The black background used dissolves its boundary and when the user press it, the action triggered is not visible.

**Expected behavior**
CloseAllTabs floating action button should have black color on a white background. It would be nice if animation could also be added to show all tabs closed as shown below.

**Steps to reproduce the behavior:**
1. Open the main screen.
2. Press tab switcher icon on the action bar.

**Screenshots**
closeallwebview

**Environment**
- Version of Kiwix Android : Build 55
- Device : Redmi 4
- OS version : Android 7.1

### PR DESCRIPTION
**Pull request by sonusourav**
Fixes #1040 

Changes:
1. Used black tint over white background in FAB.
2. Added closeAllTabs animation (cross symbol changes to tick on finish)
3. A function *openHomeScreen()* to open home screen when user closes all tabs.
( This function is similar to pull request opened by @iadeelzafar  in #1035 but with some changes, but I would like to give credits to him for this.)

Screenshots/GIF for the change: 
fab_material_design

### CODE REVIEW

**Comment by iadeelzafar** [comment]
@sonusourav Nice work on the FAB button. I'm just curious if it was necessary to redo something that has already been done?

**Comment by sonusourav** [comment]
> 
> 
> @sonusourav Nice work on the FAB button. I'm just curious if it was necessary to redo something that has already been done?

@iadeelzafar Since your pull request was not merged, so I had to do it. But still, there are some changes from yours like in my case when the user tries to close the last tab, he will be redirected to the main screen while in your case the tab-switcher is still open.

**Comment by iadeelzafar** [comment]
Since my pull request was not merged yet, you could've simply left a review on that PR instead of creating a PR for the same issue.

**Comment by sonusourav** [comment]
> 
> 
> Since my pull request was not merged yet, you could've simply left a review on that PR instead of creating a PR for the same issue. That would've been a better approach to improve Kiwix as this is how open source works.

@iadeelzafar Brother, don't get offended. I do not have any wrong intention to curb your PR. I was never even bothered about your PR. It was just a mere coincidence that whatever changes I  made fixes your issue also. Further, I have already given credit to you for this. But still not to extend this issue further, I am removing the line which says it fixes your issue.

**Comment by iadeelzafar** [comment]
@sonusourav I'm just trying to suggest what could've been a better approach. That's all.
 Happy Open Sourcing :)

**Comment by sonusourav** [comment]
@abdulwd Please review.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 1]
```diff
@@ -1150,14 +1152,46 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
       if (webViewList.size() == 0) {
         newTab(HOME_URL);
+        hideTabSwitcher();
       }
-    }, 1500);
-    updateTabSwitcherIcon();
+    }, 100);
+  }
+
+  public void rotate(FloatingActionButton v) {
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: Can you move this method to appropriate class in utils? This class is already 2k lines long. You will require `Context` which you can get from the view itself.

Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: @abdulwd I have made the changes. Please review it once.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/AnimationUtils.java [hunk 2]
```diff
@@ -61,4 +65,30 @@ public boolean willChangeBounds() {
         (int) (initialHeight / view.getContext().getResources().getDisplayMetrics().density));
     view.startAnimation(animation);
   }
+
+  //rotate animation for closeAllTab FAB
+  public static void rotate(FloatingActionButton v) {
+    float mAngleToRotate = 360f; // rotate 1 round
+    RotateAnimation wheelRotation =
+        new RotateAnimation(0.0f, mAngleToRotate, v.getWidth() / 2.0f, v.getHeight() / 2.0f);
+    wheelRotation.setDuration(50); // rotate 1 round in 50 milliSeconds
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/AnimationUtils.java]: Great work! I think the animation is very fast to be noticeable can you check the material guidelines for the duration of such animations.

File: app/src/main/java/org/kiwix/kiwixmobile/utils/AnimationUtils.java [hunk 3]
```diff
@@ -61,4 +65,30 @@ public boolean willChangeBounds() {
         (int) (initialHeight / view.getContext().getResources().getDisplayMetrics().density));
     view.startAnimation(animation);
   }
+
+  //rotate animation for closeAllTab FAB
+  public static void rotate(FloatingActionButton v) {
+    float mAngleToRotate = 360f; // rotate 1 round
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/utils/AnimationUtils.java]: Don't use the Hungarian notations checkout our code style guidelines in the repo.

**Comment by sonusourav** [comment]
@abdulwd I have made the changes, please review.

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 4]
```diff
@@ -1150,16 +1151,24 @@ public boolean onKeyDown(int keyCode, KeyEvent event) {
...
       if (webViewList.size() == 0) {
         newTab(HOME_URL);
+        hideTabSwitcher();
       }
-    }, 1500);
-    updateTabSwitcherIcon();
+    }, 300);
   }
 
+
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: Remove this line.

Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: @abdulwd Done. Thanks for reviewing.

Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: @abdulwd I have removed the line. Could you please review?

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java [hunk 5]
```diff
@@ -511,6 +514,8 @@ private void hideTabSwitcher() {
     actionBar.setDisplayShowTitleEnabled(true);
 
     drawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
+    closeAllTabsButton.setImageDrawable(
+        ContextCompat.getDrawable(getApplicationContext(), R.drawable.ic_close_white_24dp));
```
Comment (abdulwd) [inline app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: I think you can just pass `this` as the `context`.

File: app/src/main/res/values/colors.xml [hunk 6]
```diff
@@ -27,4 +27,5 @@
   <color name="blueTransparent">#962e7ac4</color>
   <color name="actionModeBackground">#4285F4</color>
   <color name="titleBar">#000000</color>
+  <color name="green_tick">#4CAF50</color>
```
Comment (abdulwd) [inline app/src/main/res/values/colors.xml]: We should follow a consistent style so change this to camel case.

Comment (sonusourav) [reply app/src/main/res/values/colors.xml]: @abdulwd But if you will see a few lines above, the style is snake case. What should I do? In fact, when I was naming it, I myself was confused which was to choose. Well according to material guidelines it should be camel case. Should I change others to camel case too?

File: app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java (hunk 5 above)
Comment (sonusourav) [reply app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java]: @abdulwd Changes done. Please review.

File: app/src/main/res/values/colors.xml (hunk 6 above)
Comment (abdulwd) [reply app/src/main/res/values/colors.xml]: Just change yours to camel case. We can change others later.

Comment (sonusourav) [reply app/src/main/res/values/colors.xml]: @abdulwd Done.

**Pull request merged**